flake.nix

```diff
@@ -10,15 +10,13 @@
     let
       pkgsFor = system: import nixpkgs {
         inherit system;
-
-        overlays = [
-          self.overlays.default
-          rust-overlay.overlays.default
-        ];
+        overlays = [ self.overlays.default ];
       };
 
       targetSystems = [ "aarch64-linux" "x86_64-linux" ];
-      mkRustToolchain = pkgs: pkgs.rust-bin.fromRustupToolchainFile ./rust-toolchain.toml;
+      mkRustToolchain = pkgs: let
+        rpkgs = pkgs.extend rust-overlay.overlays.default;
```

Comment:
```suggestion
```

I would rather pass `pkgs.extend rust-overlay.overlays.default` where necessary, like in line 24, it's an expensive operation that causes a full rebuild of nixpkgs.

For `packages` and `devShell` `rust-overlay.overlays.default` should be initialized in `pkgsFor`

Reply:
what about `lib.fix (lib.extends (import sources.rust-overlay) (self: pkgs))`. that should have less overhead than `pkgs.extend` according to https://github.com/oxalica/rust-overlay/issues/94.

Reply:
something the likes of this will work

```diff
diff --git a/flake.nix b/flake.nix
index d097383..98a8040 100644
--- a/flake.nix
+++ b/flake.nix
@@ -23,7 +23,10 @@
     {
       overlays.default = final: prev:
         let
-          rust = mkRustToolchain final;
+          rust =
+            if nixpkgs.lib.hasAttrByPath [ "rust-bin" ] final
+            then mkRustToolchain final
+            else mkRustToolchain (prev.extend rust-overlay.overlays.default);

           rustPlatform = prev.makeRustPlatform {
             cargo = rust;
```